aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+module circle::fiat_token {
+    use aptos_framework::fungible_asset::{Self, MintRef, TransferRef, BurnRef, Metadata};
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::primary_fungible_store;
+    use std::error;
+    use std::string::{Self, String};
+    use std::option;
+    use std::signer;
+    use std::vector;
+    use aptos_std::smart_table::{Self, SmartTable};
+
+    friend circle::master_minter;
+
+    /// Error codes documentation: https://aptos.dev/reference/error-codes/
+    const EALREADY_INITIALIZED: u64 = 1;
+    const EBLACKLISTED_ADDRESS: u64 = 2;
+    const ECALLER_NOT_MINTER: u64 = 3;
+    const EMINTER_ALLOWANCE_EXCEEDED: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
+    const ENOT_MASTER_MINTER: u64 = 6;
+    const ENOT_OWNER: u64 = 7;
+
+    /// Asset Symbol
+    const ASSET_SYMBOL: vector<u8> = b"USDC";
+
+    /// Module Config for Initializer
+    struct ModuleConfig has key {
+        is_initialized: bool,
+    }
+
+    /// Store refs to Manage Fiat Token
+    struct FiatTokenConfig has key {
+        mint_ref: MintRef,
+        burn_ref: BurnRef,
+        transfer_ref: TransferRef,
+    }
+
+    /// Store access info to Manage Fiat Token
+    struct FiatTokenAccessInfo has key {
+        minters: SmartTable<address, u64>,
+        blacklist: vector<address>,
```

Comment:
This should actually be a SmartVector as the blacklist can have hundreds or thousands over time right?